app/helpers/elasticsearch_helper.rb

```diff
@@ -15,47 +15,83 @@ def prefix_match(model, field, prefix, condition)
     results
   end
 
-  def taxon_search(prefix, tax_levels = TaxonCount::NAME_2_LEVEL.keys, project_id = nil)
+  def taxon_search(prefix, tax_levels = TaxonCount::NAME_2_LEVEL.keys, filters = {})
     return {} if Rails.env == "test"
     prefix = sanitize(prefix)
-    matching_taxa = {}
 
+    matching_taxa = []
+    taxon_ids = []
     tax_levels.each do |level|
```

Comment:
now that we started to use aggregations... can we get rid of the iteration over tax levels and handle that using an aggregation as well?

Reply:
Maybe... What did you have in mind?

The aggregations is to help guarantee that the genus query returns distinct results. When we were querying by genus we could have n equal results all for the same genus.
The two separate queries avoid skewing the number of results towards the species side (since there are many more of them).

Also, because we do not have an explicit level number/name, we cannot filter by a particular set of levels. What we can do is something like an additional filter like `taxid=genus_id OR taxonid=species_id ...`. I might explore that in a future PR.